include/uxr/agent/message/InputMessage.hpp

```diff
@@ -100,6 +102,22 @@ template bool InputMessage::get_payload(dds::xrce::WRITE_DATA_Payload_Data& data
 template bool InputMessage::get_payload(dds::xrce::HEARTBEAT_Payload& data);
 template bool InputMessage::get_payload(dds::xrce::ACKNACK_Payload& data);
 
+inline bool InputMessage::get_raw_payload(uint8_t* buf, size_t len)
+{
+    bool rv = false;
+    if (subheader_.submessage_length() <= len)
+    {
+        deserializer_.deserializeArray(buf, subheader_.submessage_length(), fastcdr::Cdr::BIG_ENDIANNESS);
```

Comment:
Should we capture the exception here? Or does `deseriliazeArray` return a boolean value?

Reply:
Yes, I should have captured the exception. Done!